referenceframe/transformable.go

```diff
@@ -38,34 +75,50 @@ func (pF *PoseInFrame) Name() string {
 	return pF.name
 }
 
+// SetName sets the name of the PoseInFrame.
+func (pF *PoseInFrame) SetName(name string) {
+	pF.name = name
+}
+
 // Transform changes the PoseInFrame pF into the reference frame specified by the tf argument.
 // The tf PoseInFrame represents the pose of the pF reference frame with respect to the destination reference frame.
 func (pF *PoseInFrame) Transform(tf *PoseInFrame) Transformable {
-	return NewPoseInFrame(tf.frame, spatialmath.Compose(tf.pose, pF.pose))
+	return NewPoseInFrame(tf.parent, spatialmath.Compose(tf.pose, pF.pose))
 }
 
 // NewPoseInFrame generates a new PoseInFrame.
 func NewPoseInFrame(frame string, pose spatialmath.Pose) *PoseInFrame {
 	return &PoseInFrame{
-		frame: frame,
-		pose:  pose,
+		parent: frame,
+		pose:   pose,
 	}
 }
 
 // NewNamedPoseInFrame generates a new PoseInFrame and gives it the specified name.
 func NewNamedPoseInFrame(frame string, pose spatialmath.Pose, name string) *PoseInFrame {
```

Comment:
[q] is this being used anywhere now that we have `NewLinkInFrame`? Would be good to remove if possible

Reply:
Turns out that yes this can be removed.  Have done so in my PR to your branch